src/dev/code_coverage/nyc_config/nyc.functional.config.js

```diff
@@ -7,7 +7,7 @@
  */
 
 const defaultExclude = require('@istanbuljs/schema/default-exclude');
-const extraExclude = ['data/optimize/**', 'src/core/server/**', '**/{test, types}/**/*'];
+const extraExclude = ['data/optimize/**', '**/{test, types}/**/*'];
```

Comment:
we can exclude more paths to cleanup final html report, if needed.

Reply:
@kobelb @mshustov please let me know, if you think report is missing some paths or some paths are redundant.

Current configuration does not force nyc to instrument every single file in the repo, but rather the ones **touched by test suites**. Instrumenting every single one will slow down the execution. I can check how much extra time it is adding), but we can try to configure exclusions (`packages/`, `target/` and more that nyc fails to instrument or we don't need to have.

Reply:
> Current configuration does not force nyc to instrument every single file in the repo, but rather the ones touched by test suites.

It means that we don't have test coverage for files without test suits or not executed during testing. It can skew test coverage results, so I'd rather instrument `src/core/`, `src/plugins`, and `x-pack/plugins` code.

>  Instrumenting every single one will slow down the execution. 

Let's measure and compare the actual results. It's not a big deal if it takes a few minutes longer, as we collect test coverage on CI.

> we can try to configure exclusions (packages/

Some of the packages are used in prod and must be tested(`@kbn/std`, `@kbn/server-http-tools`, to name a few). Maybe we can add include them too?

Reply:
@mshustov I pushed e95aaa5fdefd64f1fc7946b2e99e8c585f4f0674 that will collect coverage for **all files** that meet `include/exclude` conditions. So far I didn't find a quick way to instrument `packages/` and I will create a separate issue to fix it. For now it doesn't make sense to report irrelevant results (0%).
The latest report is available [here](https://kibana-coverage.elastic.dev/2022-02-03T18:12:00Z/functional-combined/index.html).